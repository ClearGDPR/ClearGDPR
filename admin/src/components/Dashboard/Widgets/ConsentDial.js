import React from 'react';

import GraphCard from 'components/core/cards/dashboard/GraphCard';
import PropTypes from 'prop-types';

const ConsentDial = ({ consented, unconsented }) => {
  const consentedPercent = consented + unconsented ? (consented + unconsented) * 100 : 100;
  return (
    <GraphCard
      title={'Consented Vs Not Consented Subjects'}
      text={`${consentedPercent}% of subjects have consented`}
      cols={2}
      data={[{ y: consented, x: 'Consented' }, { y: unconsented, x: 'Not Consented' }]}
    />
  );
};

ConsentDial.propTypes = {
  consented: PropTypes.number,
  unconsented: PropTypes.number
};

export default ConsentDial;
