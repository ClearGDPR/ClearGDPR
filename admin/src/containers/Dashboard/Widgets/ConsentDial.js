import React from 'react';
import ConsentDial from 'components/Dashboard/Widgets/ConsentDial';
import internalFetch from 'helpers/internal-fetch';
import config from 'config';
import { toast } from 'react-toastify';

export class ConsentDialContainer extends React.Component {
  state = {
    isLoading: false,
    data: {}
  };
  componentDidMount() {
    internalFetch(`${config.API_URL}/api/management/stats`)
      .then(({ data: { controller } }) => {
        this.setState({ loading: false, data: controller });
      })
      .catch(err => {
        toast.error(err.message);
        this.setState({ loading: false });
      });
  }

  render() {
    return (
      <ConsentDial
        consented={this.state.data.consented}
        unconsented={this.state.data.unconsented}
      />
    );
  }
}

export default ConsentDialContainer;
