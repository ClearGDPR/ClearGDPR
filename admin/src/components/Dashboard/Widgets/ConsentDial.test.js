import React from 'react';
import { shallow } from 'enzyme';
import ConsentDial from './ConsentDial';

describe('(Component) ConsentDial', () => {
  it('should render', () => {
    const props = {
      data: {
        consented: 1,
        unconsented: 10
      }
    };
    const component = shallow(<ConsentDial {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render if numbers are 0', () => {
    const props = {
      data: {
        consented: 0,
        unconsented: 0
      }
    };
    const component = shallow(<ConsentDial {...props} />);
    expect(component).toMatchSnapshot();
  });
});
