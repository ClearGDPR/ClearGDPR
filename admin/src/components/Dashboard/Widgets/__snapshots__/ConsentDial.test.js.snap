// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ConsentDial should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConsentDial
    data={
      Object {
        "consented": 1,
        "unconsented": 10,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "cols": 2,
      "data": Array [
        Object {
          "x": "Consented",
          "y": undefined,
        },
        Object {
          "x": "Not Consented",
          "y": undefined,
        },
      ],
      "size": 2,
      "text": "100% of subjects have consented",
      "title": "Consented Vs Not Consented Subjects",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "cols": 2,
        "data": Array [
          Object {
            "x": "Consented",
            "y": undefined,
          },
          Object {
            "x": "Not Consented",
            "y": undefined,
          },
        ],
        "size": 2,
        "text": "100% of subjects have consented",
        "title": "Consented Vs Not Consented Subjects",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`(Component) ConsentDial should render if numbers are 0 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConsentDial
    data={
      Object {
        "consented": 0,
        "unconsented": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "cols": 2,
      "data": Array [
        Object {
          "x": "Consented",
          "y": undefined,
        },
        Object {
          "x": "Not Consented",
          "y": undefined,
        },
      ],
      "size": 2,
      "text": "100% of subjects have consented",
      "title": "Consented Vs Not Consented Subjects",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "cols": 2,
        "data": Array [
          Object {
            "x": "Consented",
            "y": undefined,
          },
          Object {
            "x": "Not Consented",
            "y": undefined,
          },
        ],
        "size": 2,
        "text": "100% of subjects have consented",
        "title": "Consented Vs Not Consented Subjects",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
